### ISSUE TITLE
python input validator

### ISSUE THREAD

**Issue by iwanbk** [Jumpscale/go-raml#51]
It seems that we can use [Flask-Inputs](http://pythonhosted.org/Flask-Inputs/) for input validator.

It adds support for famous [WTForms](http://wtforms.readthedocs.org/) to validate request data.

**Comment by iwanbk** [comment]
@robvanmieghem 
For this to work, we need to create `class` for each Type and request body.

When we implement v0.3 (Possibility to generate a default server implementation to MongoDB) we will also need `class` for each Type.

I think we must use same class for both and it depends on mongodb library we will use.

**Any suggestion about which Flask/Python mongodb library we will use?**

@rkhamis @mie00

**Comment by iwanbk** [comment]
using flask-inputs is pretty easy

validation tag in this Go struct
```
type UsersPostReqBody struct {
	ID   string `json:"ID" validate:"min=4,max=8"`
	Age  int    `json:"age" validate:"min=16,max=100,multipleOf=4"`
	Item string `json:"item" validate:"min=2,regexp=^[a-zA-Z]+$"`
}
```

is equal to
```
class UserInput(Inputs):
    json = {
        #'ID': [DataRequired("ID is required")]
        'ID': [
            wtforms.validators.Length(min=4, max=8),
        ],                                        
        'age': [
            wtforms.validators.NumberRange(min=16, max=100)
        ],
        'item': [
            wtforms.validators.Length(min=2),
            wtforms.validators.Regexp('^[a-zA-Z]+$', message='invalid item format')
        ]                                                                
    }
```
custom validator also supported. We need it for multipleOf, maxItems, and minItems.

**Comment by iwanbk** [comment]
Because i can't find a way to validate nested JSON with Flask-Inputs, i tried wtforms-json.

It seems better:
- support nested JSON
- class definition is IMO more natural

```
class Address(Form):
    street = TextField(validators=[Length(min=4, max=80)])

class User(Form):
    ID = TextField("id", validators=[Length(min=4, max=8)])
    age = IntegerField("age", validators=[NumberRange(min=16, max=100)])
    item = TextField("item", validators=[Length(min=2), Regexp('^[a-zA-Z]+$')])
    address = FormField(Address, label='address')
```

I think we can use wtforms-json

**Comment by iwanbk** [comment]
fixed in #66
